client/src/components/Profile/Profile.js

```diff
@@ -0,0 +1,306 @@
+  }
+}
+
+function initialShifts() {
+  let a = [];
+  for (var i = 0; i < 168; i += 1) {
+    a.push(new Shift("#f8f8f8", null, null, null, null));
+  }
+  let count = 0;
+  for (var i = 0; i <= 23; i += 1) {
+    for (var j = 0; j <= 6; j += 1) {
+      a[count].start = i;
+      a[count].end = i + 1;
+      a[count].day = j;
+      count += 1;
+    }
+  }
+  return a;
+}
+
+var currentDate = new Date();
+var weekString =
+  format(currentDate, "MMMM") +
+  " " +
+  format(currentDate, "YYYY") +
+  ": " +
+  format(startOfWeek(currentDate), "MM/DD") +
+  " - " +
+  format(endOfWeek(currentDate), "MM/DD");
+
+export default class Profile extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { shifts: initialShifts(), schedule: [] };
+    this.currentDate = new Date();
+    this.deselectCell = <div class="deselectCell"></div>;
+    this.selectCell = <div class="selectCell"></div>;
+    this.stateFixer = this.stateFixer.bind(this);
+    this.submitClick = this.submitClick.bind(this);
+    this.cancelClick = this.cancelClick.bind(this);
+    this.renderCustomDateCell = this.renderCustomDateCell.bind(this);
+  }
+
+  componentDidMount() {
+    fetch("/availability/" + this.props.match.params.userId)
+      .then((response) => {
+        console.log("availability response");
+        return response.json();
+      })
+      .then((jsonResponse) => {
+        console.log("availability response 2");
+        this.setState({ schedule: jsonResponse.schedule });
+      });
+  }
+
+  stateFixer(e) {
+    if (
+      e.target.id != "" &&
+      e.target.style.backgroundColor != "rgb(193, 135, 211)"
+    ) {
+      /*The point of the background color check is to make sure that once a shift is requested to be covered,
+       * this can't happen again for the same shift.
+       */
+      currentClicked = e;
+      currentClickedID = e.target.id;
+      this.openModal();
+    }
+  }
+
+  renderCustomDateCell = (time, selected, innerRef) => {
+    return (
+      <div style={{ textAlign: "center" }} ref={innerRef}>
+        {selected ? this.selectCell : this.deselectCell}
+      </div>
+    );
+  };
+
+  submitClick() {
+    let reason = document.getElementById("reason");
+    let notes = reason.value;
+    fetch("/changecoverage", {
+      method: "POST",
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        coverage: true,
+        shiftID: currentClickedID,
+        sentNotes: notes,
+      }),
+    })
+      .then((response) => {
+        return response.json();
+      })
+      .then((jsonResponse) => {
+        let newShifts = this.state.shifts;
+        for (let i = 0; i < newShifts.length; i++) {
+          if (newShifts[i].id == currentClickedID) {
+            newShifts[i].color = "#C187D3";
+          }
+        }
+        this.setState({ shifts: newShifts });
+      });
+    this.closeModal();
+  }
+  cancelClick() {
```

Comment:
No modals to close either, so we can delete this too. Go through the code and make sure all the extraneous code is deleted so we don't clutter up the file.